mode-line-formatters.lisp

```diff
@@ -117,6 +117,19 @@ fmt-highlight. Any non-visible windows are colored the
   (declare (ignore ml))
   (time-format *time-modeline-string*))
 
+(add-screen-mode-line-formatter #\m 'fmt-minor-modes)
+(defun fmt-minor-modes (ml)
+  (let* ((screen (mode-line-screen ml))
+         (group (mode-line-current-group ml))
+         (head (mode-line-head ml))
+         (window (group-current-window group)))
+    (format nil "~{~A~^ ~}"
+            (loop for mode in (list-current-mode-objects
+                               :screen (mode-line-screen ml))
+                  for text = (minor-mode-lighter mode)
+                  unless (string= text "")
+                    collect text))))
```

Comment:
:thought_balloon: consider using with-output-to-string to avoid the double iteration when constructing the string